### ISSUE TITLE
Question about grpo training speed on Geo3k(sglang + megatron + qwen3.5-9B + 8*H800)

### ISSUE THREAD

**Issue by 7yzx** [verl-project/verl#7203]
Running Qwen3.5-9B multimodal GRPO on Geo3K dataset (geometry problems with images) using SGLang rollout + Megatron training backend. After tuning, steady-state is ~290s/step on 8x H800. Want to confirm whether this is expected performance or if there's room for improvement.

Steady-state per-step timing (avg of step 2-5, excluding warmup):

| Phase | Time |
|-------|------|
| gen (SGLang rollout) | ~29s |
| old_log_prob | ~84s |
| ref_log_prob | ~82s |
| update_actor | ~91s |
| update_weights | ~3.5s |
| **Total** | **~290s** |

- Throughput: ~143 tok/s
- MFU (actor update): 2.3-2.7%
- Peak GPU memory: ~59 GB / 79 GB

Environment:
| | |
|------|------|
| Python | 3.12 |
| PyTorch | 2.9.1+cu129 |
| CUDA | 12.9 / Driver 550.127.08 |
| SGLang | 0.5.10 |
| Megatron-Core | 0.16.0 |
| mbridge | 0.15.1 |
| TransformerEngine | 2.12.0 |
| Flash Attention | 2.8.3 |
| flash-linear-attention | 0.5.2 |
| transformers | 5.10.4 |
| verl | 0.9.0.dev0 |

config
```yaml
# Parallelism
actor_tp: 2, actor_pp: 1, dp: 4
rollout_tp: 8 (SGLang)
gpu_memory_utilization: 0.35

# Training
train_batch_size: 32
max_prompt_length: 1024
max_response_length: 2048
rollout_n: 5
use_dynamic_bsz: True
ppo_max_token_len_per_gpu: 4096
free_cache_engine: True
use_remove_padding: False  # required by GDN architecture
freeze_vision_tower: True

# Megatron
recompute_method: uniform
recompute_granularity: full
recompute_num_layers: 1
param/optimizer/grad_offload: all False
```

**Comment by tomhodemon** [comment]
Looking at the [wandb logs](https://wandb.ai/megatron-core-moe-dev/verl_grpo_example_geo3k/reports/qwen2-5vl-7b-veRL-megatron--VmlldzoxMzE3NjMwNg?accessToken=7oj5007dh3mr3xwlogvscgumt6dkwef3rqvfdxdhzug8jd4547mhtqprqludwibf) from the official [Algorithm Baselines](https://verl.readthedocs.io/en/latest/algo/baseline.html)' Qwen2.5-VL-7B/Geo3K run, I'd say there's room for improvement.

See the [example script](https://github.com/verl-project/verl/blob/main/examples/grpo_trainer/run_qwen2_5_vl_7b_megatron.sh) and the dedicated [Performance Tuning Guide](https://verl.readthedocs.io/en/latest/perf/perf_tuning.html).

**Comment by chinhsuanwu** [comment]
Hi! Just wondering are the performance metrics going up on your end?

<img width="647" height="321" alt="Image" src="https://github.com/user-attachments/assets/0f30da4a-b8bc-4318-9495-dc7eca04aeb7" />

I found that Qwen3.5 models aren’t improving and are actually getting worse.

**Comment by 7yzx** [comment]
> Hi! Just wondering are the performance metrics going up on your end?
> 
> <img alt="Image" width="647" height="321" src="https://private-user-images.githubusercontent.com/67839539/630201603-0f30da4a-b8bc-4318-9495-dc7eca04aeb7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODU2NDIwMjMsIm5iZiI6MTc4NTY0MTcyMywicGF0aCI6Ii82NzgzOTUzOS82MzAyMDE2MDMtMGYzMGRhNGEtYjhiYy00MzE4LTk0OTUtZGM3ZWNhMDRhZWI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA4MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwODAyVDAzMzUyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVkZmFjZDNlYjI3ZGUxMjUxNDBhZDU3MTRhNTVhYjUxNWU5NzgwM2JlOGQ1YTc4ZjI4NDYwNGJlZDI4YjhmZmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.5L-Az5GGbcPVfpGXBAWDKNkI5Nm6rVwk8OyMvtNt4c0">
> I found that Qwen3.5 models aren’t improving and are actually getting worse.

  Hi! My training metrics look normal, but I didn't configure wandb logging or evaluation for this run. By the way, this is full fine-tuning.

**critic/score/mean (reward)**

| Step | Score | Step | Score | Step | Score |
|------|-------|------|-------|------|-------|
| 1 | 0.157 | 23 | 0.540 | 45 | 0.675 |
| 5 | 0.321 | 26 | 0.585 | 50 | 0.709 |
| 10 | 0.270 | 30 | 0.506 | 53 | 0.714 |
| 11 | 0.422 | 35 | 0.591 | 55 | **0.731** |
| 16 | 0.489 | 40 | 0.613 | 60 | 0.613 |
| 20 | 0.444 | 43 | **0.754** | 65 | 0.630 |

**Comment by 7yzx** [comment]
The system failed to locate the pip-installed cuDNN library, causing conv3d to fall back to a slow kernel path. I upgrade cudnn to 9.16 solved this problem when I'm searching in xiaohongshu.

**Comment by chinhsuanwu** [comment]
Thanks so much for sharing your results!
